java/src/jmri/managers/DefaultSectionManager.java

```diff
@@ -302,41 +275,164 @@ private boolean blockSectionExists(LayoutBlock layoutBlock){
+         /*
+         Method
+         get block at siding b, and call createBlockSection
+         create sectionBlockList: set origin_block = b curr_block = b prev_block = None
+         call createSections
+         get blocks from curr_block (paths) and iterate through them
+         each iteration set nextBlock, do not use if nextBlock == prevBlock
+         get anchors of nextBlock, we want the ones which goes to the one after next
+         get the blocks either side of each anchor
+         if one of those blocks is currBlock eliminate that anchor
+         get the signal masts connected to the anchor
+         get the direction of the nextBlock, get the corresponding signal mast
+         if the signal mast exists finish, and set the entry points (i=only one because stub/siding)
+         otherwise call createSections again
+          */
+        Block layoutBlk = layoutBlock.getBlock();
+        ArrayList<Block> sectionBlockList = new ArrayList<>();
+        int index = 0;
+        Block originBlock = layoutBlk;
+        Block currBlock = layoutBlk;
+        Block prevBlock = null;
+        String indent = "  ";    //used to indent the debug statements
+        createSections(originBlock, currBlock, prevBlock, sectionBlockList, index, indent);
+    }
+
+    private void createSections(Block originBlock, Block currBlock, Block prevBlock,
+                                ArrayList<Block>sectionBlockList, int index, String indent){
+
+        index += 1;
+        // print Indent + "index", index
+        sectionBlockList.add(currBlock);
+        List <Path>paths = currBlock.getPaths();
+        //log.debug ( "{} paths {} size {} block {}", indent, paths, paths.size(), currBlock.getUserName());
         for (int j=0; j<paths.size(); j++){
             Path p = paths.get(j);
-            if (p.getBlock() != sb){
-                //this is path to an outside block, so need an Entry Point
+            Block nextBlock = p.getBlock();
+            log.debug( "{}****", indent);
+            log.debug( "{} iterating through paths {} nextBlock {}", indent, j, nextBlock.getUserName());
+            if (isNull(prevBlock) || nextBlock != prevBlock){
+                log.debug("{}", indent);
+                log.debug("{} index {} path interation {} path {} block {} ", indent, index, j, p, nextBlock.getUserName());
+                if (isNull(prevBlock)) {
+                    log.debug("{} nextBlock {} must not equal prevBlock {}", indent, nextBlock.getUserName(), "null");
+                }else{
+                    log.debug("{} nextBlock {} must not equal prevBlock {}", indent, nextBlock.getUserName(), prevBlock.getUserName());
+                }
                 String pbDir = Path.decodeDirection(p.getFromBlockDirection());
-                EntryPoint ep = new EntryPoint(sb, p.getBlock(), pbDir);
-                entryPointList.add(ep);
+                // get the panel and cutil for this Block
```

Comment:
Looking at anchor points for signal masts probably does not work since the initial outbound signal mast will frequently be at a turnout

A better approach is to look at the SML and see if the current block is a SML facing block.  if it is, you have reached the end of blocks to be included in the block based section.  Something like the following:

```
        // If the current block is a SML facing block, the next block is not needed.
        boolean checkSML = false
        for (jmri.SignalMastLogic sml : smlManager.getSignalMastLogicList()) {
            if (sml.getFacingBlock().equals(layoutBlock)) {
                checkSML = true;
                break;
            }
        }
        // setup for next loop or exit
```
This approach avoids any panel dependencies.

I don't see any limits on recursion.  You should stop when index reaches some value.

Reply:
The reason why it does not work is that to go between two stations you need a signal mast between them. There are no signal masts between leftA left b and left c  I don't think that there is any way of getting over that. The problem I was addressing was when there is one stopping point left A left B or leftc, and the next mast is anywhere, even beyond a set of turnouts. I will modify your panel to give two that work, and send them. I will also send a panel I tested on, once I have reduced the size of it.

Reply:
It doesn't seem to be working with your layout even when I remove leftA and LeftB, which is the situation I was addressing. It was working for me with a fairly complicated layout. It is late now, and I am out tomorrow. Will address it day after tomorrow....
Another problem I have noticed is that it should create sections of the form stubblockname:sectionmast, and when the stubblockname has the same name as the section mast it gets confusing.

Reply:
> The reason why it does not work is that to go between two stations you need a signal mast between them. There are no signal masts between leftA left b and left c I don't think that there is any way of getting over that. The problem I was addressing was when there is one stopping point left A left B or leftc, and the next mast is anywhere, even beyond a set of turnouts. I will modify your panel to give two that work, and send them. I will also send a panel I tested on, once I have reduced the size of it.

Sorry, I was not clear in my initial report.  The basic process of creating block based sections failed when using **Tools &rArr; Generate Sections** from the Signal Mast Logic table menu.  This has nothing to do with the Dispatcher System.

The changes here have to work for anyone using SML based sections.

The issues with the Dispatcher System are outside the scope of this activity.

Reply:
> It doesn't seem to be working with your layout even when I remove leftA and LeftB, which is the situation I was addressing. It was working for me with a fairly complicated layout. It is late now, and I am out tomorrow. Will address it day after tomorrow....
> Another problem I have noticed is that it should create sections of the form stubblockname:sectionmast, and when the stubblockname has the same name as the section mast it gets confusing.

You need to keep all of the logic based on layout blocks, blocks and paths.  These are panel independent.  

For names, my version uses the block name for one block sections and first:::last block names for multiple block sections.

Reply:
I have just checked and seen that I have omitted the essential case where there are signal masts associated with a turnout. The panel I checked on only had signal masts on anchors. The reason why I put extra blocks on the Y's of the turnouts was so that the trains would have accurate lengths for stopping, so the sections did not include a tree of turnouts in a fiddle yard. I can explain better if required. 

I have just looked briefly at your contribution. I will overlook the ethics as I owe you a pint. I will test out your version to see whether it deals with my test panel, before trying to correct mine. I expect it will. If it does, we will go with yours, as yours looks simpler. If it doesn't as I would like mine panel to work, we will toss for who does more work.

Reply:
> I have closed this branch. Dave should I delete it as well.

Yes.  The delete will remove it from you remote repository.